docs/general/ddl/data-types.rst

```diff
@@ -1233,18 +1240,17 @@ Nested object::
 ``array``
 =========
 
-CrateDB supports arrays.
-
-An array is a collection of other data types. These are:
+An array is a :ref:`compound data type <sql_ddl_datatypes_compound>` and is
+structured as a collection of other data types. Arrays can contain the
+following:
 
-* boolean
-* text
-* ip
-* all numeric types (integer, bigint, smallint, double precision, real)
-* char
-* timestamp with time zone
-* object
-* geo_point
+* :ref:`Booleans <data-type-boolean>`
+* :ref:`Character types <character-data-types>`
+* :ref:`Numeric types <data-type-numeric>`
+* :ref:`IP addresses <ip-type>`
+* :ref:`Date and time types <date-time-types>`
+* :ref:`Geographic types <sql_ddl_datatypes_geographic>`
+* :ref:`Objects <object_data_type>`
```

Comment:
Thanks for that update in order to link to the appropriate places within the documentation.

The only anomaly I see here from a doc-semantic/syntax point of view is that the reference anchor to the geographic types uses the `sql_ddl_` prefix. Maybe it is worth to adjust?

Also, while most references use dashes `-`, the reference anchors to both the geographic- and the objects- data types use underscores `_`.

Reply:
you're right. however, I am trying to limit my changes to the main docs. I am slowly updating the labels for clarity and consistency (re using "-" instead of "_")

I think it's fine to link to existing labels and not update them for consistency because if I do that, I will fall down a rabbit hole of updating labels across many pages in a way that does not directly relate to this changeset

I am only updating labels when I have to add labels to a page. i.e., when updating one or more labels in a single file, updating all the labels in that file to be consistent

if I am just linking to an existing label, I am using whatever the label currently is without going into that file to update all those labels too

thoughts?

Reply:
I am fine with anything, just wanted to point it out.